src/main/java/com/team1701/lib/util/ShooterUtil.java

```diff
@@ -0,0 +1,122 @@
+                                - shooter.getAngle().getRadians(),
+                        shooterHingePose.getRotation().getZ()));
+    }
+
+    public static Rotation2d calculateStationaryDesiredAngle(RobotState robotState) {
+        switch (robotState.getScoringMode()) {
+            case SPEAKER:
+                return Rotation2d.fromRadians(
+                        Constants.Shooter.kShooterAngleInterpolator.get(robotState.getDistanceToSpeaker()));
+            case AMP:
+                return Rotation2d.fromDegrees(Constants.Shooter.kShooterAmpAngleDegrees.get());
+            default:
+                return Rotation2d.fromRotations(Constants.Shooter.kShooterLowerLimitRotations);
+        }
+    }
+
+    public static Rotation2d calculateShooterAngleWithMotion(RobotState robotState, Translation2d expectedTranslation) {
+        return Rotation2d.fromRadians(Constants.Shooter.kShooterAngleInterpolator.get(
+                robotState.getDistanceToSpeaker(GeometryUtil.toTranslation3d(expectedTranslation))));
+    }
+
+    /**
+     * @return double[] with upper, then lower speeds
+     */
+    public static double[] calculateShooterSpeedsWithMotion(RobotState robotState, Translation2d expectedTranslation) {
+        var speed = Constants.Shooter.kShooterSpeedInterpolator.get(
+                robotState.getDistanceToSpeaker(GeometryUtil.toTranslation3d(expectedTranslation)));
+        return new double[] {speed, speed};
+    }
+
+    /**
+     * @return double[] with upper, then lower speeds
+     */
+    public static double[] calculateStationaryRollerSpeeds(RobotState robotState) {
+        switch (robotState.getScoringMode()) {
+            case SPEAKER:
+                double[] speeds = new double[2];
+                var interpolatedSpeed =
+                        Constants.Shooter.kShooterSpeedInterpolator.get(robotState.getDistanceToSpeaker());
+                for (int i = 0; i < speeds.length; i++) {
+                    speeds[i] = interpolatedSpeed;
+                }
+                Logger.recordOutput("Shooter/InterpolatedShooterSpeeds", interpolatedSpeed);
+                return speeds;
+            case AMP:
+                return new double[] {
+                    Constants.Shooter.kUpperAmpSpeed.get(), Constants.Shooter.kLowerAmpSpeed.get(),
+                };
+            default:
+                return new double[] {0, 0};
+        }
+    }
+
+    /**
+     * @return double[] with upper, then lower speeds
+     */
+    public static double[] calculateIdleRollerSpeeds(RobotState robotState, Drive drive) {
+        double[] speeds = new double[2];
+        switch (robotState.getScoringMode()) {
+            case SPEAKER:
+                double speed;
+
+                if (drive.getKinematicLimits().equals(Constants.Drive.kSlowKinematicLimits)) {
+                    speed = Constants.Shooter.kShooterSpeedInterpolator.get(robotState.getDistanceToSpeaker());
```

Comment:
I think the initial idea was that if we're driving slowly on the field, we likely intend to shoot and therefore we could ramp up higher. It doesn't take long to reach speed and I doubt we'll be driving slow mode though so it could be removed